core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -383,6 +384,16 @@ public FileInfo getFileInfo(AlluxioURI path)
       mPermissionChecker.checkPermission(FileSystemAction.READ, path);
       // getFileInfo should load from ufs if the file does not exist
       getFileId(path);
+
+      if (!mInodeTree.inodePathExists(path)) {
+        try {
+          loadMetadata(path, LoadMetadataOptions.defaults().setCreateAncestors(true));
```

Comment:
this means that if a client calls getStatus on a directory, we will always load the metadata for its content ... I think instead, this should be controlled by an option